packages/cozy-pouch-link/src/jsonapi.js

```diff
@@ -1,32 +1,40 @@
 import { generateWebLink } from 'cozy-client'
 
+const normalizeDocs = (docs, doctype, client) => {
+  for (let i = docs.length; i >= 0; i--) {
+    const doc = docs[i]
+    if (!doc) {
+      docs.splice(i, 1)
+      continue
+    }
+    normalizeDoc(doc, doctype, client)
+  }
+}
+
 export const normalizeDoc = (doc, doctype, client) => {
   const id = doc._id || doc.id
-
-  const { relationships, referenced_by } = doc
-
-  // PouchDB sends back .rev attribute but we do not want to
-  // keep it on the server. It is potentially higher than the
-  // _rev.
   const _rev = doc.rev || doc._rev
-  const normalizedDoc = {
-    ...doc,
-    id,
-    _id: id,
-    _rev,
-    _type: doctype,
-    relationships: {
-      ...relationships,
-      referenced_by
+  doc.id = id
+  doc._id = id
+  doc._rev = _rev
+  doc._type = doctype
+
+  if (doc.relationships) {
+    doc.relationships.referenced_by = doc.referenced_by
+  } else {
+    doc.relationships = {
+      referenced_by: doc.referenced_by
     }
   }
-  if (normalizedDoc.rev) {
-    delete normalizedDoc.rev
+  if (doc.rev) {
+    delete doc.rev
```

Comment:
setting to undefined is way more performant than calling delet(https://jsperf.app/delete-vs-undefined-vs-null/10 I don't like the `void 0` stuff)

Reply:
Good to know! I did the change in another PR: https://github.com/cozy/cozy-client/pull/1596/commits/2b89a635d12441ddc445cf5d44ce7ddd77232723